components/Auth.tsx

```diff
@@ -9,9 +13,12 @@ export default function Auth() {
   const handleLogin = async (email: string) => {
     try {
       setLoading(true)
-      const { user, error } = await supabaseClient.auth.signIn({ email })
-      if (error) throw error
-      alert('Check your email for the login link!')
+      const isEmailValid = await validateEmail.validate({ email }, { strict: true });
```

Comment:
I think this will error out if validation fails, so there is no need to rethrow the error.